File: clojure/src/world_news_api/specs/news.clj
(ns world-news-api.specs.news
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def news-data
  {
   (ds/opt :id) int?
   (ds/opt :title) string?
   (ds/opt :text) string?
   (ds/opt :summary) string?
   (ds/opt :url) string?
   (ds/opt :image) string?
   (ds/opt :author) string?
   (ds/opt :language) string?
   (ds/opt :source_country) string?
   (ds/opt :sentiment) float?
   })

(def news-spec
  (ds/spec
    {:name ::news
     :spec news-data}))

File: scala/src/main/scala/com/worldnewsapi/client/model/News.scala
package com.worldnewsapi.client.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import org.openapitools._
import com.worldnewsapi.client.model.BigDecimal

/**
 * A news entry.
 * @param id 
 * @param title 
 * @param text 
 * @param summary 
 * @param url 
 * @param image 
 * @param author 
 * @param language 
 * @param sourceUnderscorecountry 
 * @param sentiment 
 */
case class News(id: Option[Int],
                title: Option[String],
                text: Option[String],
                summary: Option[String],
                url: Option[String],
                image: Option[String],
                author: Option[String],
                language: Option[String],
                sourceUnderscorecountry: Option[String],
                sentiment: Option[BigDecimal]
                )

object News {
    /**
     * Creates the codec for converting News from and to JSON.
     */
    implicit val decoder: Decoder[News] = deriveDecoder
    implicit val encoder: ObjectEncoder[News] = deriveEncoder
}
